kivy/ext/mpl/backend_kivyagg.py

```diff
@@ -0,0 +1,216 @@
+from matplotlib.backends.backend_agg import FigureCanvasAgg
+from kivy.input.motionevent import MotionEvent
+
+try:
+    import kivy
+    from kivy.graphics.texture import Texture
+    from kivy.graphics import Rectangle
+    from kivy.uix.widget import Widget
+    from kivy.base import EventLoop
+except ImportError:
+    raise ImportError("this backend requires Kivy to be installed.")
+
+try:
+    kivy.require('1.9.0')
+except AttributeError:
+    raise ImportError(
+        "kivy version too old -- it must have require_version")
+
+from PIL import Image
+EventLoop.ensure_window()
+
+_debug = True
+
+
+def new_figure_manager(num, *args, **kwargs):
+    """
+    Create a new figure manager instance
+    """
+    # if a main-level app must be created, this (and
+    # new_figure_manager_given_figure) is the usual place to
+    # do it -- see backend_wx, backend_wxagg and backend_tkagg for
+    # examples. Not all GUIs require explicit instantiation of a
+    # main-level app (egg backend_gtk, backend_gtkagg) for pylab
+    FigureClass = kwargs.pop('FigureClass', Figure)
+    thisFig = FigureClass(*args, **kwargs)
+    return new_figure_manager_given_figure(num, thisFig)
+
+
+def new_figure_manager_given_figure(num, figure):
+    """
+    Create a new figure manager instance for the given figure.
+    """
+    canvas = FigureCanvasKivyAgg(figure)
+    manager = FigureManagerKivyAgg(canvas, num)
+    return manager
+
+
+class FigureCanvasKivyAgg(FigureCanvasAgg, Widget):
+    """
+    The canvas the figure renders into. Calls the draw and print fig
+    methods, creates the renderers, etc...
+
+    Public attribute
+
+      figure - A Figure instance
+
+    Note GUI templates will want to connect events for button presses,
+    mouse movements and key presses to functions that call the base
+    class methods button_press_event, button_release_event,
+    motion_notify_event, key_press_event, and key_release_event. See,
+    e.g., backend_gtk.py, backend_wx.py and backend_tkagg.py
+    """
+
+    def __init__(self, figure, **kwargs):
+        if _debug:
+            print('FigureCanvasKivyAgg: ', figure)
+        #super(FigureCanvasKivyAgg, self).__init__(figure, **kwargs)
+        FigureCanvasAgg.__init__(self, figure)
+        Widget.__init__(self, **kwargs)
+        self.figure = figure
+        self.blit()
+
+    def draw(self):
+        """
+        Draw the figure using the renderer
+        """
+        FigureCanvasAgg.draw(self)
+        if self.blitbox is None:
+            l, b, w, h = self.figure.bbox.bounds
+            w, h = int(w), int(h)
+            buf_rgba = self.get_renderer().buffer_rgba()
+        else:
+            bbox = self.blitbox
+            l, b, r, t = bbox.extents
+            w = int(r) - int(l)
+            h = int(t) - int(b)
+            t = int(b) + h
+            reg = self.copy_from_bbox(bbox)
+            buf_rgba = reg.to_string_argb()
+        texture = Texture.create(size=(w, h))
```

Comment:
You need to cache the texture and only create a new texture if the size changed. Like I do at https://github.com/matham/filers/blob/master/filers/misc_widgets.py#L267.